### ISSUE TITLE
Multiple ElasticsearchDeprecationWarning using Elasticsearch 7.7.0

### ISSUE THREAD

**Issue by yhoiseth** [wagtail/wagtail#6038]
### Issue Summary

With elasticsearch-py 7.7.0 and Elasticsearch 7.7.0, I’m getting the following deprecation warnings on `./manage.py update_index`:

```
(venv) ➜  wagtail-elasticsearch-7.7.0 ./manage.py update_index
Updating backend: default
default: Rebuilding index wagtail__wagtailcore_page
/Users/yngve/Repositories/issues/wagtail-elasticsearch-7.7.0/venv/lib/python3.7/site-packages/wagtail/search/backends/elasticsearch2.py:795: ElasticsearchDeprecationWarning: The [nGram] tokenizer name is deprecated and will be removed in a future version. Please change the tokenizer name to [ngram] instead.
  self.es.indices.create(self.name, self.backend.settings)
/Users/yngve/Repositories/issues/wagtail-elasticsearch-7.7.0/venv/lib/python3.7/site-packages/wagtail/search/backends/elasticsearch2.py:795: ElasticsearchDeprecationWarning: The [edgeNGram] tokenizer name is deprecated and will be removed in a future version. Please change the tokenizer name to [edge_ngram] instead.
  self.es.indices.create(self.name, self.backend.settings)
/Users/yngve/Repositories/issues/wagtail-elasticsearch-7.7.0/venv/lib/python3.7/site-packages/wagtail/search/backends/elasticsearch2.py:795: ElasticsearchDeprecationWarning: The [edgeNGram] token filter name is deprecated and will be removed in a future version. Please change the filter name to [edge_ngram] instead.
  self.es.indices.create(self.name, self.backend.settings)
/Users/yngve/Repositories/issues/wagtail-elasticsearch-7.7.0/venv/lib/python3.7/site-packages/wagtail/search/backends/elasticsearch2.py:795: ElasticsearchDeprecationWarning: The [nGram] token filter name is deprecated and will be removed in a future version. Please change the filter name to [ngram] instead.
  self.es.indices.create(self.name, self.backend.settings)
/Users/yngve/Repositories/issues/wagtail-elasticsearch-7.7.0/venv/lib/python3.7/site-packages/wagtail/search/backends/elasticsearch7.py:25: ElasticsearchDeprecationWarning: The [nGram] tokenizer name is deprecated and will be removed in a future version. Please change the tokenizer name to [ngram] instead.
  self.es.indices.put_mapping(index=self.name, body=mapping.get_mapping())
/Users/yngve/Repositories/issues/wagtail-elasticsearch-7.7.0/venv/lib/python3.7/site-packages/wagtail/search/backends/elasticsearch7.py:25: ElasticsearchDeprecationWarning: The [edgeNGram] tokenizer name is deprecated and will be removed in a future version. Please change the tokenizer name to [edge_ngram] instead.
  self.es.indices.put_mapping(index=self.name, body=mapping.get_mapping())
/Users/yngve/Repositories/issues/wagtail-elasticsearch-7.7.0/venv/lib/python3.7/site-packages/wagtail/search/backends/elasticsearch7.py:25: ElasticsearchDeprecationWarning: The [edgeNGram] token filter name is deprecated and will be removed in a future version. Please change the filter name to [edge_ngram] instead.
  self.es.indices.put_mapping(index=self.name, body=mapping.get_mapping())
/Users/yngve/Repositories/issues/wagtail-elasticsearch-7.7.0/venv/lib/python3.7/site-packages/wagtail/search/backends/elasticsearch7.py:25: ElasticsearchDeprecationWarning: The [nGram] token filter name is deprecated and will be removed in a future version. Please change the filter name to [ngram] instead.
  self.es.indices.put_mapping(index=self.name, body=mapping.get_mapping())
/Users/yngve/Repositories/issues/wagtail-elasticsearch-7.7.0/venv/lib/python3.7/site-packages/elasticsearch/helpers/actions.py:122: ElasticsearchDeprecationWarning: [types removal] Specifying types in bulk requests is deprecated.
  resp = client.bulk("\n".join(bulk_actions) + "\n", *args, **kwargs)
default: home.HomePage             .
default: wagtailcore.Page          .
default: indexed 2 objects

default: Rebuilding index wagtail__wagtaildocs_document
default: wagtaildocs.Document
default: indexed 0 objects

default: Rebuilding index wagtail__wagtailimages_image
default: wagtailimages.Image
default: indexed 0 objects

```

Considering that these are just deprecation warnings, we are not in a hurry to get rid of them.

### Steps to Reproduce

1. Install Elasticsearch 7.7.0
1. Clone https://github.com/yhoiseth/wagtail-elasticsearch-7.7.0 (just a fresh Wagtail project configured for Elasticsearch and with `elasticsearch-py` in `requirements.txt`)
2. `pip install -r requirements.txt`
3. `./manage.py migrate`
4. `./manage.py update_index`

* I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: yes

### Technical details

* Python version: 3.7.7
* Django version: 3.0.6
* Wagtail version: 2.9
* Browser version: Not relevant

#### Elasticsearch

```
(venv) ➜  wagtail-elasticsearch-7.7.0 git:(master) curl localhost:9200
{
  "name" : "Yngves-2015-MacBook-Pro.local",
  "cluster_name" : "elasticsearch_yngve",
  "cluster_uuid" : "xH8J69c9ToCQLKUN3cFeuA",
  "version" : {
    "number" : "7.7.0",
    "build_flavor" : "default",
    "build_type" : "tar",
    "build_hash" : "81a1e9eda8e6183f5237786246f6dced26a10eaf",
    "build_date" : "2020-05-12T02:01:37.602180Z",
    "build_snapshot" : false,
    "lucene_version" : "8.5.1",
    "minimum_wire_compatibility_version" : "6.8.0",
    "minimum_index_compatibility_version" : "6.0.0-beta1"
  },
  "tagline" : "You Know, for Search"
}
```

**Comment by yhoiseth** [comment]
Tried with Elasticsearch 7.2 and `elasticsearch-py` 7.7.0. Got only `site-packages/elasticsearch/helpers/actions.py:122: ElasticsearchDeprecationWarning: [types removal] Specifying types in bulk requests is deprecated.`.

**Comment by kaedroho** [comment]
Thanks for reporting! Looks like these are two really simple changes for the Elasticsearch 7 backend: 

 - Change occurances of `nGram` to `ngram` and `edgeNGram` to `edge_ngram`
 - Remove the `_type` parameter from the bulk request (not used in recent Elasticsearch versions anyway)

**Comment by yhoiseth** [comment]
@kaedroho Thanks for the input.

> Change occurances of `nGram` to `ngram` and `edgeNGram` to `edge_ngram`

I made a go at this in #6046. 
 
>  Remove the `_type` parameter from the bulk request (not used in recent Elasticsearch versions anyway)

This one is not obvious to me, considering that the code is in the `elasticsearch-py` repo: https://github.com/elastic/elasticsearch-py/blob/7.7/elasticsearch/helpers/actions.py#L122

Maybe I should raise the issue there instead?

**Comment by kaedroho** [comment]
We need to remove this parameter (but only for Elasticsearch 7+): https://github.com/wagtail/wagtail/blob/a501dc257d086ea4c9d38398a0cc29dc3aaa6f76/wagtail/search/backends/elasticsearch2.py#L889

The way i'd do it is wrap it in an if statement which checks an attribute on the class. For example:

```python
action = {
     '_id': mapping.get_document_id(item),
 }

if self.bulk_action_requires_type_parameter:
    action['_type'] = doc_type
```

Then on `Elasticsearch2Index` we set this to `True` but override it to `False` on `Elasticsearch7Index`

**Comment by yhoiseth** [comment]
> We just need to remove `_type` from this line:

Done.

**Comment by gasman** [comment]
Fixed in #6046
